### ISSUE TITLE
Remove <big> and <font> duplicates

### ISSUE THREAD

**Issue by robertlugg** [campbellluggblackwell/campbellluggblackwell.com#19]
In some cases, there are multiple, unnecessary tags.  Remove all but one when nested.

@ThompsonShapiro , should I change the <big> tag into something else automatically?  Sorry if you already told me.

**Comment by ThompsonShapiro** [comment]
That's not high on my priority list -- but, with one caveat, would make
rewriting those old files that were generated by Word, Excel, or SeaMonkey,
easier.

The caveat is that almost always, with only one exception I can think of,
when there's an occurrence of <xyz></xyz>, where there is nothing between
the start and end tags or if it's only "empty whitespace" that separates
them they perform no function and can be removed with no impact on the
webpage's appearance on screen.

The one exception I know of is *<p></p>*. Removing that removes a blank
line from the screen with most browsers -- which may or may not be
important. The safe thing to do is rather than remove that pair is to
replace it with *<br>*.

One thing you could do, which I hesitated to write up as an issue because
it seems petty, is a minor cleanup of the cemetery pages (which in their
genesis started out as a single, huge webpage created by Excel from a giant
spreadsheet.

As a result, many of our CampbellCemetery HTML files contain table cells
such as the following:
*<td sdval="1808" sdnum="1033;" align="RIGHT">1808</td>*

The *sdval* and *sdnum *clauses are clutter that could be removed with no
impact on the screen's appearance. They bug me, but  are harmless except
for adding a millisecond or two to the page's load time and adversely
effecting the data limits a cell phone user may have on their service
agreement. But that line could safely be changed to *<td
align="RIGHT">1808</td>*.

All the Cemetery pages need a major overhaul, which I will do in
conjunction with moving their data to MySql tables on Bluehost. But that
will probably be several months away.

I think my highest priority right now is replacing all our HTML files on
RootsWeb with the "redirect" versions. RootsWeb will probably eventually
figure out what we're doing and stop hosting the old site, but that's OK.
Those who visit while it's there will learn of the move. And the more
people who visit our new site, the more it will boost us in their rankings.
Maybe only from 10003rd to 9997th, but every bit helps.

*Family Tree Links*
One of the most helpful changes you could do next is to address changing as
many as possible family tree links to point to WikiTree rather than
wc.rootsweb. My nephew, Charles Shapiro, kindly set up my original GitHub
repository for that purpose and wrote a Linux script to change a file, the
idea being that once that worked OK a higher level script would run that
for all the HTML files in the directory. The simple table lookup I proposed
had a data problem. For reasons we can discuss later, I'd like to add an
additional column to my spreadsheet, but the first column would continue to
be the one used for doing the lookup.The "Try to replace" script should be
run periodically (or routinely). The first time it's run in production run,
most of the links to wc.rootsweb locations will be replaced. So even though
the spreadsheet contains 1,400+ rows, the number of lookups will shrink
steadily, Whenever I add someone to WikiTree (or discover they are already
there. I update the spreadsheet. Many of the folks in the tree at
wc.rootsweb are unlikely to ever be added to WikiTree

**Comment by robertlugg** [comment]
I created a new label "Low Priority" for us, which means just that.  However, some are "easy fixes", or at least I can make a partial fix easily, so I might work on them anyway.

**Comment by robertlugg** [comment]
I've created a branch with the multiple <big> tags removed.  It can be found here:

https://github.com/campbellluggblackwell/campbellluggblackwell.com/tree/big_tag_clean

**Comment by robertlugg** [comment]
CCCV1R1p02.html is a good example of multiple fonts that could be cleaned up

**Comment by ThompsonShapiro** [comment]
Thanks, I'll check it out.

Webpages containing that crap have to be redesigned & rewrtten, but your
utility simplifies that task by reducing the amount of garbage and making
it easier to see what the HTML is doing.

How feasible would it be to generalize it to removing any pair of begin and
end tags that have no intervening text? Whether it's <title></title>, <font
...></font>, or whatever. The exception being <p></p> which should be left
in place or replaced with <br>.

A bonus benefit of your utility is that the cleaned up webpages will still
be badly written,, but will load faster and for those having to pay data
rates, cheaper.. And help us stay within our allotted disk storage on
Bluehost.

**Comment by robertlugg** [comment]
Based on email discussions, We should *not* remove whitespace in text simply for the sake of trying to clean them up.  Leave as-is.

**Comment by ThompsonShapiro** [comment]
Yes, I agree. There's no shortage of work that needs to be done.

The basic philosophy of HTML was that whitespace doesn't matter. Just
provide text and let the browser decide how to display it. But of course
whitespace often does matter. So the HTML & CSS standards committees not
only had to incorporate *&nbsp;*, they had to add *margin*, *padding*,
*align-text,* <sub>, <sup>, etc. And  elements like <fig> and <figcaption>
also create whitespace.
